src/test/java/org/jabref/migrations/RenameReviewToCommentTest.java

```diff
@@ -0,0 +1,82 @@
+package org.jabref.migrations;
+
+import java.util.Collections;
+
+import org.jabref.logic.importer.ParserResult;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.FieldName;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class RenameReviewToCommentTest {
+
+    private RenameReviewToComment action;
+
+    @BeforeEach
+    public void setUp() {
+        action = new RenameReviewToComment();
+    }
+
+    @Test
+    public void noFields() {
+        BibEntry entry = createMinimalBibEntry();
+        ParserResult actualParserResult = new ParserResult(Collections.singletonList(entry));
+
+        action.performMigration(actualParserResult);
+
+        assertEquals(entry, actualParserResult.getDatabase().getEntryByKey("Entry1").get());
+    }
+
+    @Test
+    public void reviewField() {
+        BibEntry actualEntry = createMinimalBibEntry();
+        actualEntry.setField(FieldName.REVIEW, "My Review");
+        ParserResult actualParserResult = new ParserResult(Collections.singletonList(actualEntry));
+
+        BibEntry expectedEntry = createMinimalBibEntry();
+        expectedEntry.setField(FieldName.COMMENT, "My Review");
+
+        action.performMigration(actualParserResult);
+
+        assertEquals(expectedEntry, actualParserResult.getDatabase().getEntryByKey("Entry1").get());
+    }
+
+    @Test
+    public void commentField() {
+        BibEntry entry = createMinimalBibEntry();
+        entry.setField(FieldName.COMMENT, "My Comment");
+        ParserResult actualParserResult = new ParserResult(Collections.singletonList(entry));
+
+        action.performMigration(actualParserResult);
+
+        assertEquals(entry, actualParserResult.getDatabase().getEntryByKey("Entry1").get());
+    }
+
+
+    @Test
+    public void reviewAndCommentField() {
```

Comment:
I am not sure whether we should keep the two fields.

Maybe, I am storing the comments of reviewers in the `review` field? -- OK, then I should put it into the `reviews` (with `s`) field... 

I would let the user do it manually then. We have the "Other fields" tab, where "Review" will show up then, don't we?

Reply:
Fine with all of it. I think, as usual, I am discussing an edge-case here. JabRef never merged field contents in its entire history. As far as I remember, I cannot disable this single migration, can I? If not, JabRef changes data in a more hard way as it did in the past.

Since no other software (hopefully!) does not use the "review" field, it is OK for me to merge it into the comment field. Let's hope for the best... In case there is software relying on this field, JabRef will remove it... OK, the user can always skip the migration...

Reply:
Actually, with the current implementation the user cannot skip the migration. It will be executed each times any database is opened!

To clarify:
Goal: Have a prominent position for users to make comments on their entries.

Opinionated observation: The Review Tab is less important than a Comments tab, a review is basically also a comment on the entry.

Proposed change: Rename the Review Tab to Comments Tab, consolidate fields. If both fields present merge.

Yes, this is a hard change, but what's the alternative? Keeping everything as it is/Close PR? Add another Tab? Swap the two fields? Make the migration optional? Let user choose the migration strategy? Discuss at devcall?

Reply:
How about just popping up a dialog IF a review field is found in the database? Just a dialog that informs the user that the review field will be appended to the comment field and removed from the entries. If the user does not want this, she should not save the file.

Good compromise imho and it informs the users what is actually going on.

Reply:
I was only discussing the impact if **both** fields are present. The other silent migration (if only `review` is there) was OK for me. I like, however, the idea of @lenhard. 

I would like to have that dialog consistent with the old one at "Upgrade file"

![grafik](https://user-images.githubusercontent.com/1366654/36229333-2ca7e01e-11d7-11e8-8374-06ff5e6a8573.png)

Reply:
I think a simple "the review field has been merged with the comments field" information is enough. If the user does not like our decision, then he is free to use other software :-)

Reply:
@tobiasdiez And even the information should not be a dialog, but some kind of notification:

![grafik](https://user-images.githubusercontent.com/1366654/36243397-9abd0864-1221-11e8-8f49-0507231ee5ea.png)

Maybe, this is is a task for #CloudRef. I sometimes have the feeling, we recode a UI framework in JabRef. Would it have been easier (in the long run) if we have used IntelliJ's UI as basis? Or some NetBeans UI designer (recommended by https://www.b-tu.de/fg-software-systemtechnik/team/mitarbeiter/markus-uhlig) or even the modern Eclipse thing? OK, JavaFX is way more modern than these :)

Reply:
or just use controlsfx as we do: https://controlsfx.bitbucket.io/org/controlsfx/control/NotificationPane.html or https://controlsfx.bitbucket.io/org/controlsfx/control/Notifications.html

Reply:
@tobiasdiez The problem is the following: This is called before the JabRefFrame is created (where I would put the code for the Notification).
So where should I create such a Notification? Just creating an Object anywhere doesn't work, since just writing
```
Platform.runLater(() ->Notifications.create().title("Title Text").text("Hello World 0!"));
```
does not display anything.

Reply:
Yeah, I think for the moment it best to just show a normal dialog. With the move to a javafx frame we should be able to use notifications in the future...

Reply:
Okay did that. You can even reject the merge now... however this will pop up again the next time one opens an unmigrated database.

Reply:
> I think a simple "the review field has been merged with the comments field" information is enough. If the user does not like our decision, then he is free to use other software :-)

Fundamentally, I don't see the problem with a spare field a user can choose to use or to ignore; but for users who have been using both fields for different types of data, this is a huge inconvenience. 

(either to manually edit every entry so the fields remain separate; or move to a new software; or keep an older version of jabref running)

As well as knocking my faith in the program going forwards (what fields can I rely on in the future?). 

& especially because a database manager isn't like a word processor (i.e.) where you're constantly creating new files, so you can have some .DOCs and some .DOCXs and it's not a big deal. A database is a longterm object, it needs to remain predictable over its lifespan and archive data reliably. Updating the software shouldn't lead to data loss (which is essentially what this is - I had data recorded in a field, that field no longer exists, and the data is no longer filed effectively).

The appeal of the separate Comment and Review fields *is* that it's purpose is uncertain - I can't appropriate the author field, say, for a new use because it's too likely that the author field will be useful for its original purpose. But a field with no clear function is extremely useful, as each user will likely have some unique need database designers cannot predict. The concept of a Review (specific data reviewing the entry in some way) and the concept of a Comment (any extra miscellaneous data to record with the entry) is distinct. 

To me this feels like removing a feature needlessly, without checking how people are using it.

(I've been using Jabref for over 6 years and only just updated my laptop, so I'm only now trying the new version - but that's a lot of data and wasted time, or difficulty and risk to keep an old laptop chugging along, or try and run an older version, or - I don't even know. The current users are the people who think the program is already good and have invested trust and time into it; and trustworthiness is an essential quality of a database, moreso than other kinds of program.)

Reply:
You can use the custom entry types dialog to fully customize your entries (also reflected in the entry editor) and add a review field yourself.

Reply:
@TarryDan Sorry, this issue is causing you inconvenience.  What you could do to work around it for now is to open your (old) database with a [text editor](https://alternativeto.net/software/notepad-plus-plus/?license=opensource) and use the "find and replace" functionality to rename the review fields into a different field of your choice. In recent versions of JabRef, it is also possible to conveniently transfer field content to different fields via the automatic field editor (Edit > Automatic field editor > copy or move content), so if there ever were to happen another migration in future, it would be much easier.